javalib/src/main/scala/java/nio/ByteArrayBits.scala

```diff
@@ -0,0 +1,181 @@
+package java.nio
+
+private[nio] object ByteArrayBits {
+  def apply(array: Array[Byte], arrayOffset: Int,
+      isBigEndian: Boolean, indexMultiplier: Int = 1): ByteArrayBits =
+    new ByteArrayBits(array, arrayOffset, isBigEndian, indexMultiplier)
+}
+
+@inline
+private[nio] final class ByteArrayBits(
```

Comment:
This thing deserves a comment why it does everything with tuples instead of lists/arrays. (I assume inlining). Also, it might be a good idea to annotate methods with `@inline` you expect to be inlined for documentation purposes, even if the optimizer would do so anyway.